exp/api/v1beta1/azuremanagedcontrolplane_types.go

```diff
@@ -206,6 +206,9 @@ type ManagedControlPlaneVirtualNetwork struct {
 	CIDRBlock string `json:"cidrBlock"`
 	// +optional
 	Subnet ManagedControlPlaneSubnet `json:"subnet,omitempty"`
+	// ResourceGroupName is the name of the Azure resource group for the VNet and Subnet.
+	// +optional
+	ResourceGroupName string `json:"resourceGroupName,omitempty"`
```

Comment:
```suggestion
	ResourceGroup string `json:"resourceGroup,omitempty"`
```

To be consistent with other usages in the API (AzureCluster.Spec, VNetSpec, etc.)

Reply:
ack

Reply:
https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/68f0b75dcd0cfe7e6f82008f16e41f1a78b9ce41/exp/api/v1beta1/azuremanagedcontrolplane_types.go#L45
I actually copied it from here to keep it consistent with this 😅

Reply:
I did a quick audit, and the value of `ResourceGroupName` under `AzureManagedControlPlaneSpec` is the single exception in the capz type system; the common convention is to name the property `ResourceGroup`, and to include a method `ResourceGroupName()` that returns the value of that string property. (I'm not sure of the value of that `ResourceGroupName()` method, but I digress...

I agree with @CecileRobertMichon that we should prefer `ResourceGroup` in this PR, and we will probably schedule a breaking change to the `AzureManagedControlPlaneSpec` type struct to remove this exception:

- https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/2701